rnd/autogpt_server/autogpt_server/app.py

```diff
@@ -13,13 +16,28 @@ def get_config_and_secrets():
     return settings
 
 
+@retry(stop=stop_after_attempt(30), wait=wait_exponential(multiplier=1, min=1, max=30))
+def wait_for_nameserver():
+    pyro.locate_ns(host="localhost", port=9090)
+    print("NameServer is ready")
+
+
 def run_processes(processes: list[AppProcess], **kwargs):
     """
     Execute all processes in the app. The last process is run in the foreground.
     """
     try:
-        for process in processes[:-1]:
+        # Start NameServer first
+        processes[0].start(background=True, **kwargs)
```

Comment:
pyro is not great for a k8s environment so I'm having issues with it not starting correctly everytime. I've put this in to ensure to wait that the NS has started before other processes.

Reply:
Since we have been retrying on locate_ns, can't we simply wait longer?
I see this as doing the same as retrying longer.

Reply:
I can test it out and see

Reply:
I've tried different configurations for waiting longer and longer but it seems to still get stuck in a continous loop on gke. I can take a a deeper look at what's going on as part of another ticket, but to unblock - I'm gonna merge as this set up does _magically_ work 😅  - I agree with you there's no gaurantee but I'm having issues getting other work on top of this deploying

Reply:
🥲 okay, but can you at least call 
```
PyroNameServer.start(background=True, **kwargs)
wait_for_nameserver()
```

in main instead of modifying the run_processes ? should be exactly the same thing without breaking the run_processes

Reply:
```
    PyroNameServer().start(background=True)
    wait_for_nameserver()

    run_processes(
        [
            # Remove PyroNameServer() here
            ExecutionManager(),
            ExecutionScheduler(),
            AgentServer(),
        ],
        **kwargs
    )
```

Reply:
actually nvm, it's fine, the name server will be killed anyway, right